src/vaults/vaults.utils.ts

```diff
@@ -463,6 +464,29 @@ function constructGraphVaultData(
   });
 }
 
+export async function estimateHarvestEventApr(
+  chain: Chain,
+  vault: VaultDefinition,
+  start: number,
+  end: number,
+  amount: VaultPerformanceEvent['amount'],
+  balance: number,
+): Promise<number> {
+  const duration = end - start;
+
+  const depositToken = await getFullToken(chain, vault.depositToken);
+
+  const fmtBalance = formatBalance(balance, depositToken.decimals);
+
+  const measuredBalance = fmtBalance / duration;
+  const totalHarvestedTokens = formatBalance(amount || BigNumber.from(0), depositToken.decimals);
```

Comment:
so since the token won't always be the same denomination when the method is processing a tree distribution we will need to calculate the value of these tokens 

then we can use the next piece as you have but using liek 

`totalHarvestedTokensValue / balanceValue) * durationScalar * 100 (human readable)`